### PR TITLE
EVG-19163: Configure GraphQL code generation

### PR DESCRIPTION
**Pull request by minnakt**
EVG-19163

### Description 
This PR creates the codegen config and adds the `check_codegen` task. Also cleaned up the YAML file a bit.

### Testing 
- `check_codegen` task should run and pass

### CODE REVIEW

File: .evergreen.yml [hunk 1]
```diff
@@ -443,8 +459,6 @@ tasks:
 
   - name: e2e_test
     commands:
-    - func: get-evergreen-project
-    - func: get-logkeeper-project
```
Comment (sophstad) [inline .evergreen.yml:447]: Sweet 😎

File: .evergreen.yml [hunk 2]
```diff
@@ -230,6 +239,24 @@ functions:
...
+        ln -s evergreen/graphql/schema sdlschema
+
+  yarn-codegen:
+    command: shell.exec
+    params:
+      working_dir: parsley
+      shell: bash
+      script: |
+        ${PREPARE_SHELL}
+        chmod +x ./scripts/check_codegen.sh
```
Comment (sophstad) [inline .evergreen.yml:257]: If we ensure the file is considered executable by git, is this necessary? (more info [here](https://stackoverflow.com/questions/40978921/how-to-add-chmod-permissions-to-file-in-git))

File: .evergreen.yml
Comment (sophstad) [inline .evergreen.yml:1]: While you're here you could remove the display_name fields under `attach-storybook` 🤣

**Review by sophstad — APPROVED** [review]

File: .evergreen.yml (hunk 2 above)
Comment (minnakt) [reply .evergreen.yml]: gave it a try, but maybe something about Evergreen makes it not work 🤔

**Pull request merged**